src/EoS/Nuclear/readtable.cpp

```diff
@@ -109,60 +120,95 @@ void nuc_eos_C_ReadTable( char *nuceos_table_name )
 
 
 // read size of tables
-   READ_EOS_HDF5( "pointsrho",    &g_nrho,  H5T_NATIVE_INT, H5S_ALL );
-   READ_EOS_HDF5( "pointsenergy", &g_neps,  H5T_NATIVE_INT, H5S_ALL );
-   READ_EOS_HDF5( "pointsye",     &g_nye,   H5T_NATIVE_INT, H5S_ALL );
-   READ_EOS_HDF5( "points_mode",  &g_nmode, H5T_NATIVE_INT, H5S_ALL );
-
+   READ_EOS_HDF5( "pointsrho",      &g_nrho,      H5T_NATIVE_INT, H5S_ALL );
+#if   ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+   READ_EOS_HDF5( "pointstemp",     &g_ntemp,     H5T_NATIVE_INT, H5S_ALL );
+#elif ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+   READ_EOS_HDF5( "pointsenergy",   &g_neps,      H5T_NATIVE_INT, H5S_ALL );
+#endif
+   READ_EOS_HDF5( "pointsye",       &g_nye,       H5T_NATIVE_INT, H5S_ALL );
+   READ_EOS_HDF5( "pointsrho_mode", &g_nrho_mode, H5T_NATIVE_INT, H5S_ALL );
+   READ_EOS_HDF5( "points_mode",    &g_nmode,     H5T_NATIVE_INT, H5S_ALL );
+   READ_EOS_HDF5( "pointsye_mode",  &g_nye_mode,  H5T_NATIVE_INT, H5S_ALL );
+
+
+#if   ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+   int n_def_mode = g_ntemp;
+#elif ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+   int n_def_mode = g_neps;
+#endif
+ 
 
 // allocate memory for tables
    real *alltables_temp      = NULL;
    real *alltables_mode_temp = NULL;
```

Comment:
Would it be better to rename these two `alltables` variabeles since it is no longer for temperature only?

Reply:
sorry for the confusement this alltabless_temp means temporary array for alltables. Later we change the indices and store in alltables

Reply:
How about using `_tmp`?

Reply:
sounds great.